buildenv/jenkins/jobs/infrastructure/copyrightCheck

```diff
@@ -446,3 +469,24 @@ def checkMatches(checkFile, fileExtensions, filePaths) {
     }
     return false;
 }
+
+def get_files() {
+    return sh (
+        script: "git diff -C --diff-filter=ACM --name-only origin/${ghprbTargetBranch} HEAD",
+        returnStdout: true
+        ).trim()
+}
+
+def convert_text_file(inFile) {
+    def fileType = sh (
+        script: "file -b ${inFile}",
+        returnStdout: true
+    ).trim()
+
+    if (fileType && fileType.contains('EBCDIC text')) {
+        sh """
+            iconv -f IBM-1047 -t ISO8859-1 ${inFile} > ${inFile}.new
+            mv ${inFile}.new ${inFile}
```

Comment:
Overwriting the files doesn't seem a good idea.

Reply:
Recreate the file.

Reply:
My comment wasn't about how the file was overwritten, but that we are overwriting it with different content. Is there another way to do this that doesn't involve replacing the file? Can we read the file and then convert it in memory?

Reply:
readFile with encoding could work

Reply:
Replaced with readFile with encoding.